applications/chats/services.py

```diff
@@ -0,0 +1,35 @@
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+
+from .models import Room
+
+
+class RoomCreator:
+
+    @transaction.atomic
+    def create_from_target_match(self, target, *members):
+        if self._exists_equal_room(target.topic, *members):
+            return None
+
+        room = Room.objects.create(topic=target.topic)
+
+        self._validate_members(room, *members)
+
+        room.members.add(*members)
```

Comment:
This looks like a good place to use [get_or_create](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#get-or-create).

Reply:
`members` param is a user list that already exists and I'm adding to the room member set

Reply:
Yeah, I was talking about the whole `create_from_target_match` function. You could use `get_or_create` to do what you want because if it already exists it does nothing.